### PR TITLE
Add getDiffFiles tests

### PR DESCRIPTION
**Pull request by val-fed**

### CODE REVIEW

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+  private Protos.DiffFilesResponse getResponse(String workspace, long diffNumber) {
+    Protos.DiffFilesRequest request =
+        Protos.DiffFilesRequest.newBuilder().setWorkspace(workspace).setDiffId(diffNumber).build();
+
+    return blockingStub.getDiffFiles(request);
+  }
+
+  @Test
+  public void testFirestoreClientMock() {
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Why do you need this test?

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+//    workspaceCommand.run(args);
+
+    when(firestoreClient.getDocument(
+            "reviewer/data/diff/2", Protos.DiffFilesResponse.newBuilder()))
+        .thenReturn(getExpectedResponse(testFileCommitId));
+    when(firestoreClient.getDocument(
+        "/reviewer/data/last_diff_id", Protos.DiffNumberResponse.newBuilder()))
+        .thenReturn(Protos.DiffNumberResponse.newBuilder().setLastDiffId(1).build());
+
+    component.getDiffCommand().run(new String[] {"diff"});
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Why do you need the `DiffCommand` when we're testing `CodeReviewService,GetDiffFiles()`?

File: tools/reviewer/service/tests/TestTool.java [hunk 3]
```diff
@@ -99,9 +100,24 @@ public void runGetFile() {
...
+  }
+
+  public void runGetDiffFiles(String workspace, long diffNumber) {
+    System.out.println(getDiffFiles(workspace, diffNumber));
+  }
+
   public static void main(String[] args) {
     TestTool tool = new TestTool();
-    tool.runGetTextDiff();
+    tool.runGetDiffFiles("ws", 65);
```
Comment (oferb) [inline tools/reviewer/service/tests/TestTool.java]: `ws` is the workspace name? So it's at `base/ws/ws`?

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 1 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: It just to check that "mocking" works. I'll delete it.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 2 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Deleted it.

File: tools/reviewer/service/tests/TestTool.java (hunk 3 above)
Comment (oferb) [reply tools/reviewer/service/tests/TestTool.java]: Ohh, just noticed this is for the `TestTool`. Nevermind then.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+                  }
+                })
+            .build();
+    gitRepoFactory = component.getGitRepoFactory();
+    fileUtils = component.getFileUtils();
+
+    String projectId = component.getAuthService().getProjectId();
+    String token = component.getAuthService().getToken();
+    when(firestoreClientFactory.create(projectId, token)).thenReturn(firestoreClient);
+    firestoreClientFactory.create(projectId, token);
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Why do you need to call `firestoreClientFactory.create(projectId, token);`?

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+                    return aaBaseFolder;
+                  }
+                })
+            .build();
+    gitRepoFactory = component.getGitRepoFactory();
+    fileUtils = component.getFileUtils();
+
+    String projectId = component.getAuthService().getProjectId();
+    String token = component.getAuthService().getToken();
+    when(firestoreClientFactory.create(projectId, token)).thenReturn(firestoreClient);
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Instead of getting the `projectId` and `token`, can you use `any()` instead?
https://stackoverflow.com/questions/1778744/using-mockitos-generic-any-method

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+            .directExecutor()
+            .addService(codeReviewService)
+            .build()
+            .start();
+    channel = InProcessChannelBuilder.forName(serverName).directExecutor().build();
+    blockingStub = CodeReviewServiceGrpc.newBlockingStub(channel);
+  }
+
+  private String joinPaths(String first, String... more) {
+    return FileSystems.getDefault().getPath(first, more).toAbsolutePath().toString();
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Why not use `fileUtils`?

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+    Protos.DiffFilesRequest request =
+        Protos.DiffFilesRequest.newBuilder()
+            .setWorkspace(TEST_WORKSPACE)
+            .setDiffId(DIFF_ID)
+            .build();
+
+    return blockingStub.getDiffFiles(request);
+  }
+
+  private com.google.startupos.common.repo.Protos.BranchInfo getExpectedBranchInfo() {
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: It's better to import `BranchInfo` instead. That way all imports are at the top and someone looking there can get an idea of what this file is using.
Similarly with other protos.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+                        .setWorkspace("ws1")
+                        .setRepoId(REPO_ID)
+                        .setCommitId(testFileCommitId)
+                        .setFilenameWithRepo("startup-os/test_file.txt")
+                        .build())
+                .build())
+        .build();
+  }
+
+  private Protos.DiffFilesResponse getExpectedResponseAddingCommit(
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: I think it would be nicer to import `DiffFilesResponse` so you don't need the `Protos.` prefix.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 9]
```diff
@@ -0,0 +1,450 @@
...
+    codeReviewService =
+        new CodeReviewService(
+            component.getAuthService(),
+            fileUtils,
+            aaBaseFolder,
+            gitRepoFactory,
+            component.getTextDifferencer(),
+            firestoreClientFactory);
+
+    createInitialRepo(initialRepoFolder);
```
Comment (oferb) [inline tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Can you maybe order the methods a little bit more? There's a lot of them and it's a little hard to follow the code.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 4 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Deleted it.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 5 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Fixed.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 6 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Done.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 7 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Done.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 8 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Done.

File: tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java (hunk 9 above)
Comment (val-fed) [reply tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Moved in order of use.

**Comment by val-fed** [comment]
[Add getDiffFiles tests](https://trello.com/c/7Nq0ztJ4/20-add-getdifffiles-tests)

File: tools/reviewer/local_server/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 10]
```diff
@@ -0,0 +1,435 @@
...
+    repo.init();
+    repo.setFakeUsersData();
+    fileUtils.writeStringUnchecked(
+        TEST_FILE_CONTENTS, fileUtils.joinPaths(initialRepoFolder, FILE_IN_HEAD));
+    fileInHeadCommitId = repo.commit(repo.getUncommittedFiles(), "Initial commit").getId();
+  }
+
+  private void initAaBase(String initialRepoFolder, String aaBaseFolder) {
+    InitCommand initCommand = component.getInitCommand();
+    InitCommand.startuposRepo.resetValueForTesting();
```
Comment (oferb) [inline tools/reviewer/local_server/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Why do you need to reset value? It doesn't get overridden?

Comment (val-fed) [reply tools/reviewer/local_server/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Actually, this line isn't necessary. Deleted it. 
Also I deleted the same line in `CodeReviewServiceTextDiffTest`.

File: tools/reviewer/local_server/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 11]
```diff
@@ -0,0 +1,435 @@
...
+    initCommand.run(args);
+  }
+
+  private void createAaWorkspace(String name) {
+    WorkspaceCommand workspaceCommand = component.getWorkspaceCommand();
+    String[] args = {"workspace", "-f", name};
+    workspaceCommand.run(args);
+    repoPath = fileUtils.joinPaths(getWorkspaceFolder(TEST_WORKSPACE), "startup-os");
+    repo = gitRepoFactory.create(repoPath);
+    repo.setFakeUsersData();
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/reviewer/local_server/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Done.

File: tools/reviewer/local_server/service/tests/CodeReviewServiceGetDiffFilesTest.java [hunk 12]
```diff
@@ -0,0 +1,435 @@
...
+    aaBaseFolder = fileUtils.joinToAbsolutePath(testFolder, "base_folder");
+
+    when(firestoreClientFactory.create(any(), any())).thenReturn(firestoreClient);
+
+    codeReviewService =
+        new CodeReviewService(
+            component.getAuthService(),
+            fileUtils,
+            aaBaseFolder,
+            gitRepoFactory,
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/reviewer/local_server/service/tests/CodeReviewServiceGetDiffFilesTest.java]: Done.
